### PR TITLE
fix: Clone template modals

### PR DESCRIPTION
**Pull request by cyaiox**
This PR creates a new CloneTemplate modal and updates the `duplicateProject` action to handle better the loading states.

image

### CODE REVIEW

File: src/lib/api/builder.ts [hunk 1]
```diff
@@ -219,7 +219,9 @@ function fromRemoteProject(remoteProject: RemoteProject): Project {
     id: remoteProject.id,
     title: remoteProject.title,
     description: remoteProject.description,
-    thumbnail: `${BUILDER_SERVER_URL}/projects/${remoteProject.id}/media/thumbnail.png`,
+    thumbnail: remoteProject.is_template
```
Comment (cyaiox) [inline src/lib/api/builder.ts:222]: This change is to load the template thumbnail image from the public folder. 

same for video.

File: src/modules/project/actions.ts [hunk 2]
```diff
@@ -68,12 +68,17 @@ export type DeleteProjectAction = ReturnType<typeof deleteProject>
 
 // Duplicate project
 
-export const DUPLICATE_PROJECT = 'Duplicate project'
+export const DUPLICATE_PROJECT_REQUEST = '[Request] Duplicate project'
```
Comment (cyaiox) [inline src/modules/project/actions.ts:71]: Updated this actions to handle better the duplication project flow

File: src/modules/project/sagas.ts [hunk 3]
```diff
@@ -130,42 +132,46 @@ export function* projectSaga(builder: BuilderAPI) {
...
     let thumbnail: string = project.thumbnail
 
-    if (isTemplatesEnabled && project.isTemplate) {
-      thumbnail = yield call(getImageAsDataUrl, project.thumbnail)
-    } else if (thumbnail && isRemoteURL(thumbnail)) {
-      thumbnail = yield call(getImageAsDataUrl, project.thumbnail)
-    }
+    try {
+      if ((thumbnail && isRemoteURL(thumbnail)) || (isTemplatesEnabled && project.isTemplate)) {
+        thumbnail = yield call(getImageAsDataUrl, project.thumbnail)
```
Comment (cyaiox) [inline src/modules/project/sagas.ts:145]: if the project is a `template`, it's possible to load the thumbnail from the public folder and send it to the backend as base64.

File: src/components/SceneCreationSelector/SceneCreationSelector.tsx [hunk 4]
```diff
@@ -19,15 +21,15 @@ export default function SceneCreationSelector({ onOpenModal, onNavigate }: Props
...
+        imgSrc={`${PUBLIC_URL}/images/scratch-preview-img.svg`}
         description={t('scenes_page.no_scenes.from_scratch.description')}
       />
       <SceneCard
         onClick={handleCreateFromTemplate}
         title={t('scenes_page.no_scenes.from_template.title')}
-        videoSrc="/videos/template-preview.mp4"
-        imgSrc="/images/template-preview-img.svg"
+        videoSrc={`${PUBLIC_URL}/videos/template-preview.mp4`}
+        imgSrc={`${PUBLIC_URL}/images/template-preview-img.svg`}
```
[the remark this answers is not in the document]
Comment (cyaiox) [reply src/components/SceneCreationSelector/SceneCreationSelector.tsx]: thanks for the suggestion! updated 🙌

**Review by flobarreto — APPROVED** [review]

**Pull request merged**